sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,241 @@ use crate::{
-                        &outgoing_connections,
-                        map,
-                        &mut forked_rng,
-                    )
-                },
+        // bound probabilities to ensure we're getting some diveristy of agents
+        let lower_bound_prob = 0.05;
+        let upper_bound_prob = 0.90;
+        let prob_local_resident = if workers_cap == 0 {
+            lower_bound_prob
+        } else {
+            f64::min(
+                upper_bound_prob,
+                f64::max(lower_bound_prob, residents_cap as f64 / num_trips as f64),
             )
-            .into_iter()
-            .flatten()
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
+        };
+        let prob_local_worker = f64::min(
+            upper_bound_prob,
+            f64::max(lower_bound_prob, workers_cap as f64 / num_trips as f64),
+        );
+
+        info!(
+            "BUILDINGS - workplaces: {}, residences: {}, mixed: {}",
+            num_bldg_commercial, num_bldg_residential, num_bldg_mixed_residential_commercial
+        );
+        info!(
+            "CAPACITY - workers_cap: {}, residents_cap: {}, prob_local_worker: {}, \
+             prob_local_resident: {}",
+            workers_cap, residents_cap, prob_local_worker, prob_local_resident
+        );
+
+        let mut num_trips_local = 0;
+        let mut num_trips_commuting_in = 0;
+        let mut num_trips_commuting_out = 0;
+        let mut num_trips_passthru = 0;
+        timer.start("create people");
+        for _ in 0..num_trips {
+            let (is_local_resident, is_local_worker) = (
+                rng.gen_bool(prob_local_resident),
+                rng.gen_bool(prob_local_worker),
+            );
+            let from_home = if is_local_resident {
+                if let Some(residence) = residents.pop() {
+                    TripEndpoint::Bldg(residence)
+                } else {
+                    warn!(
+                        "unexpectedly out of residential capacity, falling back to off-map \
+                         residence"
+                    );
+                    TripEndpoint::Border(map.all_incoming_borders().choose(rng).unwrap().id, None)
+                }
+            } else {
+                // MJK: investigate what is offmap location about?
+                TripEndpoint::Border(map.all_incoming_borders().choose(rng).unwrap().id, None)
+            };
+
+            let mut build_offmap_commute = || {
+                (
+                    // TODO: currently we treat all borders as viable commutes. We should prefer
+                    // larger thoroughfares to reflect reality.
+                    TripEndpoint::Border(map.all_outgoing_borders().choose(rng).unwrap().id, None),
+                    // TODO: for the return trip, we should prefer an incoming border *near* the
+                    // outoing trip, rather than a completely random border.
```

Comment:
@dabreegster - is there any existing mechanism  for finding an *incoming* border *near* a given outgoing border?

Reply:
Sometimes the same intersection is both incoming and outgoing. `Intersection` has `is_incoming_border`, `is_outgoing_border`. For "near", the "as the bird flies" answer is `FindClosest`. For "near" as in distance of an actual route, it's more complicated, because you have to pick a mode, and if it's a complicated map with things like access-restricted zones, also a destination. The quick approximation could just ignore one-ways, lane types, etc and do something like `path_dist_to` in `raw.rs`.

Reply:
excellent advice. Thank you!